src/include/lean/lean.h

```diff
@@ -1927,6 +1927,70 @@ static inline size_t lean_data_hashmap_mk_idx(lean_object* sz, uint64_t hash) {
     return (size_t)(hash & (lean_unbox(sz) - 1));
 }
 
+static inline size_t lean_data_hashset_mk_idx(lean_object* sz, uint64_t hash) {
+    return (size_t)(hash & (lean_unbox(sz) - 1));
+}
+
+static inline uint64_t lean_expr_binderinfo_to_uint64(uint8_t binderinfo) {
+  return (uint64_t)binderinfo;
+}
+
+static inline uint64_t lean_expr_data(lean_object *expr) {
+    return lean_ctor_get_uint64(expr, lean_ctor_num_objs(expr)*sizeof(void*));
+}
+
+static inline lean_object* lean_expr_checker_get_max_ctor_fields(lean_obj_arg _unit) {
+    return lean_box(LEAN_MAX_CTOR_FIELDS);
+}
+
+static inline lean_object* lean_expr_checker_get_max_ctor_scalars_size(lean_obj_arg _unit) {
+    return lean_box(LEAN_MAX_CTOR_SCALARS_SIZE);
+}
+
+static inline lean_object* lean_expr_checker_get_usize_size(lean_obj_arg _unit) {
+    return lean_box(sizeof(size_t));
+}
+
+static inline lean_object* lean_expr_checker_get_max_ctor_tag(lean_obj_arg _unit) {
+    return lean_box(LeanMaxCtorTag);
+}
+
+static inline uint8_t lean_init_core_strict_or(uint8_t b1, uint8_t b2) {
+    return b1 || b2;
+}
+
+static inline uint8_t lean_init_core_strict_and(uint8_t b1, uint8_t b2) {
+    return b1 && b2;
+}
+
+static inline lean_object *lean_init_meta_version_get_major(lean_obj_arg _unit) {
+    return lean_box(LEAN_VERSION_MAJOR);
+}
+
+static inline lean_object *lean_init_meta_version_get_minor(lean_obj_arg _unit) {
+    return lean_box(LEAN_VERSION_MINOR);
+}
+
+static inline lean_object *lean_init_meta_version_get_patch(lean_obj_arg _unit) {
+    return lean_box(LEAN_VERSION_PATCH);
+}
+
+static inline uint8_t lean_init_meta_version_is_release(lean_obj_arg _unit) {
+    return LEAN_VERSION_IS_RELEASE;
+}
+
+static inline lean_object *lean_init_meta_version_get_special_desc(lean_obj_arg _unit) {
+    return lean_mk_string(LEAN_SPECIAL_VERSION_DESC);
+}
+
+static inline uint8_t lean_init_meta_internal_is_stage0(lean_obj_arg _unit) {
+    return LEAN_IS_STAGE0;
+}
```

Comment:
I do not think the prefixes (`expr_checker`, `init_core`. `init_meta`, `init_meta_internal`) are useful for these. The constants they are wrapping do not have the prefix and such prefixing is not the style used in other C externs.

Reply:
@tydeu note that the fully qualified name `lean_data_hashmap_mk_idx` already exists (https://github.com/leanprover/lean4/blob/master/src/include/lean/lean.h#L1926-L1928). I followed suit and decided to fully qualify the new ones I have outlines. 

I am happy to remove the prefixes.

Reply:
@bollu Hmm. I think that case is an outlier (and probably should be fixed), other externs do not have module prefixes. See, for instance, other externs in [`Init.Core`](https://github.com/leanprover/lean4/blob/53477089fe255280d7a816371ec315c6e2e94220/src/Init/Core.lean#L370-L371) and [`Init.IO`](https://github.com/leanprover/lean4/blob/53477089fe255280d7a816371ec315c6e2e94220/src/Init/System/IO.lean#L169). Afaik, the standard style is for C symbols to only include *namespace* prefixes, not *module* ones.